tests/components/bayesian/test_binary_sensor.py

```diff
@@ -686,3 +693,79 @@ async def test_reload(hass):
 
 def _get_fixtures_base_path():
     return path.dirname(path.dirname(path.dirname(__file__)))
+
+
+async def test_template_triggers(hass):
+    """Test sensor with template triggers."""
+    hass.states.async_set("input_boolean.test", STATE_OFF)
+    config = {
+        "binary_sensor": {
+            "name": "Test_Binary",
+            "platform": "bayesian",
+            "observations": [
+                {
+                    "platform": "template",
+                    "value_template": "{{ states.input_boolean.test.state }}",
+                    "prob_given_true": 1999.9,
+                },
+            ],
+            "prior": 0.2,
+            "probability_threshold": 0.32,
+        }
+    }
+
+    await async_setup_component(hass, "binary_sensor", config)
+    await hass.async_block_till_done()
+
+    assert hass.states.get("binary_sensor.test_binary").state == STATE_OFF
+
+    events = []
+    hass.helpers.event.async_track_state_change_event(
+        "binary_sensor.test_binary", callback(lambda event: events.append(event))
+    )
+
+    context = Context()
+    hass.states.async_set("input_boolean.test", STATE_ON, context=context)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    assert events[0].context == context
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)